src/System.Private.CoreLib/shared/System/IO/PersistedFiles.Unix.cs

```diff
@@ -100,7 +101,7 @@ internal static string GetHomeDirectory()
                 const int BufLen = Interop.Sys.Passwd.InitialBufferSize;
                 byte* stackBuf = stackalloc byte[BufLen];
                 if (TryGetHomeDirectoryFromPasswd(stackBuf, BufLen, out userHomeDirectory))
-                    return userHomeDirectory;
+                    return userHomeDirectory!; // TODO-NULLABLE: https://github.com/dotnet/roslyn/issues/26761
```

Comment:
This can still be null even if `TryGetHomeDirectoryFromPasswd` returned true, it only returns false if the buffer was to small.

Reply:
Good point.  I've removed the TODO.